packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_user_client.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { IUserUiSettingsClient } from '@kbn/core-ui-settings-server/src';
+import type { UiSettingsServiceOptions } from '../types';
+import { BaseUiSettingsClient } from './base_ui_settings_client';
+
+interface UserProvidedValue<T = unknown> {
+  userValue?: T;
+  isOverridden?: boolean;
+}
+
+type UserProvided<T = unknown> = Record<string, UserProvidedValue<T>>;
+
+/**
+ * Common logic for setting / removing keys in a {@link IUserUiSettingsClient} implementation
+ */
+export class UiSettingsUserClient extends BaseUiSettingsClient implements IUserUiSettingsClient {
+  private readonly userProfileSettingsClient: UiSettingsServiceOptions['userProfileSettingsClient'];
+  private readonly request?: KibanaRequest;
+
+  constructor(options: UiSettingsServiceOptions, request?: KibanaRequest) {
+    super(options);
+    const { userProfileSettingsClient } = options;
+    this.userProfileSettingsClient = userProfileSettingsClient;
+    this.request = request;
+  }
+
+  async getUserProfileSettings(): Promise<Record<string, string>> {
```

Comment:
I am now passing in the `request` to the `getUserProfileSettings` function based on other code review feedback, by adding the request as an argument, I would break the interface.